http-server-netty/src/test/groovy/io/micronaut/http/server/netty/cors/CorsFilterSpec.groovy

```diff
@@ -456,85 +306,34 @@ class CorsFilterSpec extends Specification {
-        originConfig.allowedMethods = [HttpMethod.GET]
-        originConfig.allowedHeaders = ['foo', 'bar']
-
-        HttpServerConfiguration.CorsConfiguration config = enabledCorsConfiguration([foo: originConfig])
-
-        CorsFilter corsHandler = buildCorsHandler(config)
-
-        when:
-        Optional<MutableHttpResponse<?>> result = filterOk(corsHandler, request)
-
-        then:
-        result.isPresent()
+        HttpRequest request = HttpRequest.create(HttpMethod.OPTIONS, "/doesnt-exists-route")
+                .header(HttpHeaders.ORIGIN, 'http://www.foo.com')
+                .header(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD, 'GET')
 
         when:
-        MutableHttpResponse<?> response = result.get()
+        HttpResponse<?> response = execute(request)
 
         then:
         HttpStatus.OK == response.status()
     }
 
+    @Property(name = "micronaut.server.cors.configurations.foo.allowedOrigins", value = "http://www.foo.com")
+    @Property(name = "micronaut.server.cors.configurations.foo.exposedHeaders", value = "Foo-Header,Bar-Header")
     void "test preflight handleRequest on route that does exist but doesn't handle requested HTTP Method"() {
         given:
-
-        CorsOriginConfiguration originConfig = new CorsOriginConfiguration()
-        originConfig.exposedHeaders = ['Foo-Header', 'Bar-Header']
-
-        HttpServerConfiguration.CorsConfiguration config = enabledCorsConfiguration([foo: originConfig])
-
-        CorsFilter corsHandler = buildCorsHandler(config)
-
-        String origin = 'http://www.foo.com'
-        HttpHeaders headers = Stub(HttpHeaders) {
-            getOrigin() >> Optional.of(origin)
-            getFirst(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD, _) >> Optional.of(HttpMethod.POST)
-            contains(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD) >> true
-        }
-        URI uri = new URIBuilder( '/example' ).build()
-        HttpRequest request = Stub(HttpRequest) {
-            getHeaders() >> headers
-            getMethod() >> HttpMethod.OPTIONS
-            getUri() >> uri
-        }
-
-        List<UriRouteMatch<?,?>> routes = embeddedServer.getApplicationContext().getBean(Router).
-                findAny(request)
-        request.getAttribute(HttpAttributes.AVAILABLE_HTTP_METHODS, _) >> Optional.of(routes.stream().map(route->route.getHttpMethod()).collect(Collectors.toList()))
+        HttpRequest request = HttpRequest.create(HttpMethod.OPTIONS, "/example")
+                .header(HttpHeaders.ORIGIN, 'http://www.foo.com')
+                .header(HttpHeaders.ACCESS_CONTROL_REQUEST_METHOD, 'POST')
 
         when:
-        Optional<MutableHttpResponse<?>> result = filterOk(corsHandler, request)
+        HttpResponse<?> response = execute(request)
 
         then:
-        result.isPresent()
-
-        when:
-        MutableHttpResponse<?> response = result.get()
-
-        then:
-        HttpStatus.OK == response.status()
+        HttpStatus.FORBIDDEN == response.status()
```

Comment:
I had to rewite the mocked test. This is not passing anymore, the mock didn't include the proper origin mapping so the request filter was skipped. Please check how it supposted to work @yawkat @sdelamo